test/view.js

```diff
@@ -0,0 +1,397 @@
+    });
+
+    it('removes item from view as conditions are NO longer satisfied', function() {
+      var model    = (new Model).at('_page');
+      
+      model = addData(model); // Add items to collection
+      model.fn('yellowFruits', function(emit, fruit) {
+        if (fruit.color === 'yellow') {
+          emit(fruit.name + '*' + fruit.color + '*' + fruit.amount*2, '_page.fruits.' + fruit.id);
+        }
+      });
+
+      // Before updating item included in view
+      var view = model.at('fruits').view('yellowFruits');
+      view.ref('_page.filteredFruits');
+      var expectedFruits = createExpectedResult(model, ['bananaId', 'lemonId']);
+      expect(model.get('filteredFruits')).to.eql(expectedFruits);
+     
+      model.set('fruits.bananaId.color', 'green'); // Update item
+
+      // After updating the item, it is now excluded from the view
+      view = model.at('fruits').view('yellowFruits');
+      view.ref('_page.filteredFruits');
+      var expectedFruits = createExpectedResult(model, ['lemonId']);
+      expect(model.get('filteredFruits')).to.eql(expectedFruits);
+    });
+  });
+});
+
+describe('Derby-View: Events', function() {
+    describe('Adding new item to collection', function() {
+      it('View is empty: triggers "change" event', function() {
+        var model    = (new Model).at('_page');
+
+        model.fn('yellowFruits', function(emit, fruit) {
+          if (fruit.color === 'yellow') {
+            emit(fruit.name + '*' + fruit.color + '*' + fruit.amount*2, '_page.fruits.' + fruit.id);
+          }
+        });
+        var view = model.at('fruits').view('yellowFruits');
+        view.ref('_page.filteredFruits');
+
+        model.on('change', '**', addListener); // Set up event listener
+        function addListener(segments, value, previous, passed) {
+          var fruitId = value.id;
+          expect(model.get('fruits.' + fruitId)).to.eql(value);
+        }
+        // Add new item to empty collection
+        model.add('fruits', {name: 'grapefruit', color: 'orange', amount: 10, id: 'grapefruitId'});
+         // Add new item to non-empty collection
+         model.add('fruits', {name: 'banana', color: 'yellow', amount: 15, id: 'bananaId'});
+      });
+
+      it('View is NOT empty: triggers "change" event', function() {
+        var model    = (new Model).at('_page');
+
+        model.fn('yellowFruits', function(emit, fruit) {
+          if (fruit.color === 'yellow') {
+            emit(fruit.name + '*' + fruit.color + '*' + fruit.amount*2, '_page.fruits.' + fruit.id);
+          }
+        });
+        var view = model.at('fruits').view('yellowFruits');
+        view.ref('_page.filteredFruits');
+
+        model.on('change', '**', addListener); // Set up event listener
+        function addListener(segments, value, previous, passed) {
+            var fruitId = value.id;
+            expect(model.get('fruits.' + fruitId)).to.eql(value);
```

Comment:
Another tell that you are not necessarily testing the proper things is that you're relying on the processing data too much, and not specifying it enough yourself. This could cause tests to pass while things are not working properly. For example, what happens if only one event is triggered in this test? (it will still pass).